File: src/main/java/lazy/helper/tileentity/syncable/ISyncableObject.java
package lazy.helper.tileentity.syncable;

import java.io.DataInput;
import java.io.DataOutput;
import java.io.IOException;

import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.world.World;

public interface ISyncableObject {
	/*

	public void readFromStream(DataInput stream) throws IOException;

	public void writeToStream(DataOutput stream, boolean fullData) throws IOException;
	*/

	public void writeToNBT(NBTTagCompound nbt, String name);

	public void readFromNBT(NBTTagCompound nbt, String name);

	public void markDirty();
	
	public boolean isDirty();

	public void markClean();
}

File: src/main/scala/geek/lawsof/physics/lib/objects/block/nbt/traits/ISyncableObject.scala
package geek.lawsof.physics.lib.objects.block.nbt.traits

import net.minecraft.nbt.NBTTagCompound

/**
 * Created by anshuman on 28-05-2014.
 */
trait ISyncableObject {
  def writeToNBT(nbt: NBTTagCompound, name: String)

  def readFromNBT(nbt: NBTTagCompound, name: String)

  var dirty = false

  def markDirty = dirty = true

  def markClean = dirty = false
}
